packages/astro/src/core/cookies/cookies.ts

```diff
@@ -120,7 +120,9 @@ class AstroCookies implements AstroCookiesInterface {
 		const values = this.#ensureParsed(options);
 		if (key in values) {
 			const value = values[key];
-			return new AstroCookie(value);
+			if (value) {
+				return new AstroCookie(value);
+			}
```

Comment:
The new type returns `Record<string, string | undefined>`, so to avoid possible breakages in `AstroCookie`, I create the type only when the value isn't `undefined`